# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1257979934 18000
#      Wed Nov 11 17:52:14 2009 -0500
# Node ID 965487e0028249b947e7a76feac65141ec829a89
# Parent  873ec09e387a380b450bfdd3f5c1e14c9eeb7dfe
stairs.m (__stairs__): correct nargin check; new demos

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-11-11  John W. Eaton  <jwe@octave.org>
+
+	* plot/stairs.m (__stairs__): Correct nargin check.  New demos.
+	From Jakub Kasse <jakub.kasse@tul.cz>.
+
 2009-11-11  John W. Eaton  <jwe@octave.org>
 
 	* help/help.m (help): Return immediately after calling do_contents.
 	(do_contents): Find all matching directories, not just the first.
 	Call get_help_text_from_file to avoid looking in path.
 	Handle printing here.
 
 2009-11-10  John W. Eaton  <jwe@octave.org>
diff --git a/scripts/plot/stairs.m b/scripts/plot/stairs.m
--- a/scripts/plot/stairs.m
+++ b/scripts/plot/stairs.m
@@ -75,17 +75,17 @@ function [xs, ys] = stairs (varargin)
     if (nargout == 1)
       xs = h;
     endif
   endif
 endfunction
 
 function [h, xs, ys] = __stairs__ (doplot, varargin)
 
-  if (nargin == 1 || ischar (varargin{2}))
+  if (nargin == 2 || ischar (varargin{2}))
     y = varargin {1};
     varargin(1) = [];
     if (ismatrix (y))
       if (isvector (y))
 	y = y(:);
       endif
       x = 1:rows (y);
     endif
@@ -212,16 +212,23 @@ endfunction
 %! stairs (x, y);
 
 %!demo
 %! x = 1:10;
 %! y = rand (1, 10);
 %! [xs, ys] = stairs (x, y);
 %! plot (xs, ys);
 
+%!demo
+%! stairs (1:9);
+
+%!demo
+%! [xs, ys] = stairs (9:-1:1);
+%! plot (xs, ys);
+
 function update_props (h, d)
   set (get (h, "children"), "color", get (h, "color"), 
        "linewidth", get (h, "linewidth"),
        "linestyle", get (h, "linestyle"),
        "marker", get (h, "marker"),
        "markerfacecolor", get (h, "markerfacecolor"),
        "markeredgecolor", get (h, "markeredgecolor"),
        "markersize", get (h, "markersize"));
